library/swift/src/ResponseHandler.swift

```diff
@@ -66,8 +66,9 @@ public final class ResponseHandler: NSObject {
     @escaping (_ error: EnvoyError) -> Void)
     -> ResponseHandler
   {
-    self.underlyingCallbacks.onError = { errorCode, message in
-      closure(EnvoyError(errorCode: errorCode, message: message, cause: nil))
+    self.underlyingCallbacks.onError = { errorCode, message, attemptCount in
+      closure(EnvoyError(errorCode: errorCode, message: message,
+                         attemptCount: attemptCount < 0 ? nil : UInt32(attemptCount), cause: nil))
```

Comment:
You need to use `UInt32(exactly: attemptCount)` here to prevent crashes in case `attemptCount` is out of bounds. If you want, you could then additionally remove the ternary, as `UInt32(exactly: attemptCount)` will also return `nil` if the value is negative, but you may also want to keep it for clarity 🤷